overlord/devicestate/handlers.go

```diff
@@ -144,12 +139,22 @@ func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) (err error) {
 		}
 	}()
 
-	// only useful for testing
+	// this will be set only in tests
 	if injectedSetModelError != nil {
 		return injectedSetModelError
 	}
 
-	return remodCtx.Finish()
+	// add the assertion only after everything else was successful
+	err = assertstate.Add(st, new)
+	if err != nil && !isSameAssertsRevision(err) {
+		return err
+	}
+	// and finish (this will set the new model)
+	if err := remodCtx.Finish(); err != nil {
```

Comment:
any reason not to keep this as return remodCtx.Finish() without the if (with the comment). Also for the same model/brand finish is a nop, so it just return nil, which is good

Reply:
Sorry, no reason. I moved back to the previous style.